src/haddock/libs/libsubprocess.py

```diff
@@ -229,7 +243,34 @@ def run(
                         remove_original=True,
                     )
 
-        if error:
-            raise CNSRunningError(error)
-
+        # If undetected error or detect an error in the STDOUT
+        if error or self.contains_cns_stdout_error(out):
+            # Write .err file
+            with open(self.error_file, "wb+") as errf:
+                errf.write(out)
+            # Compress it
+            if compress_err:
+                gzip_files(self.error_file, remove_original=True)
+            if error:
+                raise CNSRunningError(error)
+
+        # Return STDOUT
         return out
+
+    @staticmethod
+    def contains_cns_stdout_error(out: bytes) -> bool:
+        # Decode end of STDOUT
+        # Search in last 24000 characters (300 lines * 80 characters)
+        sout = out[-24000:].split(bytes(os.linesep, "utf-8"))
+        print(sout)
+        print(type(sout))
+        print()
+        # Reverse loop on lines (read backward)
+        for bytes_line in reversed(sout):
+            line = bytes_line.decode("utf-8")
+            if "         ^^^^^^^" in line:
+                return True
+            # Check if a known error is found
```

Comment:
Add a comment here saying that this check is also needed because known errors will not have error markers